### PR TITLE
[TW-1558] Create overview for Mock an API

### PR DESCRIPTION
**Pull request by mark-anthony-postman**
Adds a new overview at `/docs/designing-and-developing-your-api/mocking-data/mock-an-api/`

### CODE REVIEW

File: src/pages/docs/designing-and-developing-your-api/mocking-data/mock-an-api.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+Once you have your collection, you're ready to [create a new mock server](/docs/designing-and-developing-your-api/mocking-data/setting-up-mock/). You can then [make calls to the mock server](/docs/designing-and-developing-your-api/mocking-data/setting-up-mock/#making-requests-to-mock-servers) from Postman or from a client application. The mock server simulates your API's behavior, enabling your to test your API or develop new functionality before your API is production ready.
+
+> If needed, the mock server can [generate dynamic responses](/docs/designing-and-developing-your-api/mocking-data/creating-dynamic-responses/) using variables and templates, enabling you to better simulate real-world behavior.
+
+## Level up with step-by-step guides
+
+If you need more help setting up a mock server, follow the step-by-step guide to [mocking your API using saved response examples](/docs/designing-and-developing-your-api/mocking-data/mocking-with-examples/). You'll learn how to add requests and examples to your collection, then use the collection to configure and use a mock server.
```
Comment (MatthewPinsker-postman) [inline src/pages/docs/designing-and-developing-your-api/mocking-data/mock-an-api.md:49]:
```suggestion
If you need more help setting up a mock server, follow the step-by-step guide to [mock your API using saved response examples](/docs/designing-and-developing-your-api/mocking-data/mocking-with-examples/). You'll learn how to add requests and examples to your collection, then use the collection to configure and use a mock server.
```

**Review by MatthewPinsker-postman — APPROVED** [review]
LGTM! I added one minor suggestion.

File: src/pages/docs/designing-and-developing-your-api/mocking-data/mock-an-api.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+* [Get started with mock servers in Postman](#get-started-with-mock-servers-in-postman)
+* [Level up with step-by-step guides](#level-up-with-step-by-step-guides)
+* [Learn how Postman mock servers work](#learn-how-postman-mock-servers-work)
+
+## Get started with mock servers in Postman
+
+Before you set up a new mock server, create a [collection](/docs/sending-requests/intro-to-collections/) with the API requests you want to mock, then add [saved examples](/docs/sending-requests/examples/) to each request. The examples have the data you want the mock server to return in response to each request.
+
+Once you have your collection, you're ready to [create a new mock server](/docs/designing-and-developing-your-api/mocking-data/setting-up-mock/). You can then [make calls to the mock server](/docs/designing-and-developing-your-api/mocking-data/setting-up-mock/#making-requests-to-mock-servers) from Postman or from a client application. The mock server simulates your API's behavior, enabling your to test your API or develop new functionality before your API is production ready.
```
Comment (nerminamiller-postman) [inline src/pages/docs/designing-and-developing-your-api/mocking-data/mock-an-api.md:43]:
```suggestion
Once you have your collection, you're ready to [create a new mock server](/docs/designing-and-developing-your-api/mocking-data/setting-up-mock/). You can then [make calls to the mock server](/docs/designing-and-developing-your-api/mocking-data/setting-up-mock/#making-requests-to-mock-servers) from Postman or from a client application. The mock server simulates your API's behavior, enabling your to test your API or develop new functionality before your API is production ready.
```

File: src/pages/docs/designing-and-developing-your-api/mocking-data/mock-an-api.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+> If needed, the mock server can [generate dynamic responses](/docs/designing-and-developing-your-api/mocking-data/creating-dynamic-responses/) using variables and templates, enabling you to better simulate real-world behavior.
+
+## Level up with step-by-step guides
+
+If you need more help setting up a mock server, follow the step-by-step guide to [mock your API using saved response examples](/docs/designing-and-developing-your-api/mocking-data/mocking-with-examples/). You'll learn how to add requests and examples to your collection, then use the collection to configure and use a mock server.
+
+You can also get step-by-step help for [creating mock servers using the Postman API](/docs/designing-and-developing-your-api/mocking-data/mock-with-api/) if you want to automate mock server creation.
```
Comment (nerminamiller-postman) [inline src/pages/docs/designing-and-developing-your-api/mocking-data/mock-an-api.md:51]:
```suggestion
You can also automate mock server creation by following step-by-step help for [creating mock servers using the Postman API](/docs/designing-and-developing-your-api/mocking-data/mock-with-api/).
```

**Review by nerminamiller-postman — APPROVED** [review]
Really nice writeup, Mark! Left you just a couple of suggestions.

File: src/pages/docs/designing-and-developing-your-api/mocking-data/mock-an-api.md (hunk 3 above)
Comment (mark-anthony-postman) [reply src/pages/docs/designing-and-developing-your-api/mocking-data/mock-an-api.md:51]: Yes, this was a bit awkward. Building on your suggestion, I went with:
> If you want to automate your mock servers, follow the step-by-step guide to set up mock servers using the Postman API. You'll learn how to use the Postman API to mock a collection and get the mock server URL.

File: src/pages/docs/designing-and-developing-your-api/mocking-data/mock-an-api.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    name: "Case Studies"
+  - type: link
+    name: "The Trevor Project uses mock servers to understand API responses early"
+    url: "https://www.postman.com/case-studies/the-trevor-project/"
+  - type: link
+    name: "OVO accelerated integration implementation with mock servers"
+    url: "https://www.postman.com/case-studies/ovo/"
+---
+
+A mock server enables you to simulate your API without having to configure a real API server. You can use a Postman Collection to set up a mock server. When you send a request to the mock server, Postman returns a real-world response using data from your collection.
```
Comment (jkonrath-postman) [inline src/pages/docs/designing-and-developing-your-api/mocking-data/mock-an-api.md:31]:
```suggestion
A _mock server_ enables you to simulate your API without having to configure a real API server. You can use a Postman Collection to set up a mock server. When you send a request to the mock server, Postman returns a real-world response using data from your collection.
```

**Review by jkonrath-postman — APPROVED** [review]
LGTM!

**Pull request merged**